### PR TITLE
Updating the Python readme

### PR DESCRIPTION
**Pull request by lupesko**
## Description ##
Updating the readme with:
- Updated links
- Extended description
- Unit tests pre-requisite

No code changes.

@aaronmarkham @nswamy @sandeep-krishnamurthy

### CODE REVIEW

File: python/README.md [hunk 1]
```diff
@@ -3,20 +3,23 @@ MXNet Python Package
...
+For running unit tests, you will need the [nose PyPi package](https://pypi.python.org/pypi/nose). To install:
+```bash
+pip install nose
+```
+
+Once ```nose``` is installed, run the following from MXNet root directory:
 
 ```
 nosetests tests/python/unittest
 nosetests tests/python/train
```
Comment (pracheer) [inline python/README.md]: Blank line not needed.

File: python/README.md [hunk 2]
```diff
@@ -3,20 +3,23 @@ MXNet Python Package
 MXNet is a deep learning framework designed for both *efficiency* and *flexibility*.
 It allows you to mix the flavours of deep learning programs together to maximize the efficiency and your productivity.
```
Comment (pracheer) [inline python/README.md]: nitpick: I think this MXNet intro seems out of place here. If required, we can do something like this:
"This directory and nested files contain [MXNet's](https://github.com/apache/incubator-mxnet) Python package and language binding."

File: python/README.md [hunk 3]
```diff
@@ -3,20 +3,23 @@ MXNet Python Package
...
 
+This directory and nested files contain MXNet Python package and language binding.
 
-Installation
-------------
-To install, check [Build Instruction](http://mxnet.io/get_started/install.html)
+## Installation
```
Comment (sandeep-krishnamurthy) [inline python/README.md]: Installation link should be 
https://mxnet.incubator.apache.org/install/index.html

File: python/README.md (hunk 2 above)
Comment (lupesko) [reply python/README.md]: Agreed and done.

File: python/README.md (hunk 3 above)
Comment (lupesko) [reply python/README.md]: Fixed.

File: python/README.md (hunk 1 above)
Comment (lupesko) [reply python/README.md]: Fixed.

**Comment by lupesko** [comment]
@sandeep-krishnamurthy @pracheer comments addressed.

**Pull request merged**